### ISSUE TITLE
nuxt 3 Hybrid Rendering build did not exit

### ISSUE THREAD

**Issue by BrowserScan** [nuxt/nuxt#26502]
### Environment

Nuxt project info:                                                                                                                        10:25:43 AM

------------------------------
- Operating System: Linux
- Node Version:     v18.19.1
- Nuxt Version:     3.11.1
- CLI Version:      3.11.1
- Nitro Version:    2.9.4
- Package Manager:  yarn@1.22.19
- Builder:          -
- User Config:      debug, runtimeConfig, modules, site, sitemap, ogImage, gtag, build, app, css, experimental, vite, nitro, srcDir, typescript, devServer
- Runtime Modules:  @nuxtjs/seo@2.0.0-rc.10, nuxt-og-image@3.0.0-rc.47, nuxt-gtag@0.5.9, nuxt-scheduler@0.1.9
- Build Modules:    -
------------------------------

### Reproduction

nux.config.ts

```js
{
debug: true,
ssr: true,
nitro: {
        rollupConfig: {
            plugins: process.env.NODE_ENV === "production" ? [replacePlugin] : [],
        },
       // not exit when configure the prerender routers
        prerender: {
            crawlLinks: false,
            routes: ['/sitemap.xml'],
            failOnError: true,
        },
    },
}
```

### Describe the bug

ssr with prerender not done, must be ctrl + C , and no error
```bash
  ├─ .output/server/index.mjs (426 B) (241 B gzip)
  └─ .output/server/package.json (4.12 kB) (1.4 kB gzip)
Σ Total size: 45.7 MB (19.8 MB gzip)
[nitro] compiled: 0.965ms
✔ You can preview this build using node .output/server/index.mjs                                                                   nitro 10:05:47 AM
[nuxt] build:done: 57.817s                                                                                                                10:05:47 AM
[nitro] close: 0.29ms                                                                                                                     10:05:47 AM
[nuxt] close: 0.929ms                                                                                                                     10:05:47 AM
^C
```

only ssr 
```bash
  ├─ .output/server/index.mjs (426 B) (241 B gzip)
  └─ .output/server/package.json (4.12 kB) (1.4 kB gzip)
Σ Total size: 45.7 MB (19.8 MB gzip)
[nitro] compiled: 0.59ms
✔ You can preview this build using node .output/server/index.mjs                                                                   nitro 10:08:26 AM
[nuxt] build:done: 27.356s                                                                                                                10:08:26 AM
[nitro] close: 0.216ms                                                                                                                    10:08:26 AM
[nuxt] close: 0.655ms                                                                                                                     10:08:26 AM
Done in 79.57s.
```

### Additional context

I can't continue the CI-CD process without exiting

### Logs

```shell-script
[nitro] compiled: 0.035ms                                                                                                                 10:05:03 AM
ℹ Prerendering 1 routes                                                                                                            nitro 10:05:03 AM
[nitro-runtime] request: 0.023ms                                                                                                          10:05:03 AM
[nitro-runtime] site-config:init: 0.025ms                                                                                                 10:05:03 AM
[nitro-runtime] beforeResponse: 0.021ms                                                                                                   10:05:03 AM
[nitro-runtime] afterResponse: 0.023ms                                                                                                    10:05:03 AM
[nitro] prerender:generate: 0.396ms                                                                                                       10:05:03 AM
[nitro] prerender:route: 0.025ms                                                                                                          10:05:03 AM
  ├─ /__sitemap__/style.xsl (17ms)                                                                                                  nitro 10:05:03 AM
[nitro] rollup:before: 0.04ms                                                                                                             10:05:03 AM
[nitro] types:extend: 0.028ms  
...
[nitro] compiled: 0.023ms                                                                                                                 10:05:15 AM
[nitro-runtime] request: 0.028ms                                                                                                          10:05:15 AM
[nitro-runtime] site-config:init: 0.02ms                                                                                                  10:05:15 AM
[nitro-runtime] sitemap:resolved: 0.025ms                                                                                                 10:05:15 AM
[nitro-runtime] sitemap:output: 0.02ms                                                                                                    10:05:15 AM
[nitro-runtime] beforeResponse: 0.019ms                                                                                                   10:05:15 AM
[nitro-runtime] afterResponse: 0.017ms                                                                                                    10:05:15 AM
  ├─ /sitemap.xml (29ms)                                                                                                            nitro 10:05:15 AM
[nitro] prerender:done: 12.309s                                                                                                           10:05:15 AM
ℹ Prerendered 2 routes in 26.272 seconds                                                                                           nitro 10:05:15 AM
✔ Generated public .output/public                                                                                                  nitro 10:05:16 AM
[nuxt] nitro:build:public-assets: 0.263ms
[nitro] rollup:before: 214.089ms
[nitro] types:extend: 0.028ms
ℹ Building Nuxt Nitro server (preset: node-server)                                                                                 nitro 10:05:16 AM
✔ Nuxt Nitro server built 
...
  ├─ .output/server/index.mjs (426 B) (241 B gzip)
  └─ .output/server/package.json (4.12 kB) (1.4 kB gzip)
Σ Total size: 45.7 MB (19.8 MB gzip)
[nitro] compiled: 0.965ms
✔ You can preview this build using node .output/server/index.mjs                                                                   nitro 10:05:47 AM
[nuxt] build:done: 57.817s                                                                                                                10:05:47 AM
[nitro] close: 0.29ms                                                                                                                     10:05:47 AM
[nuxt] close: 0.929ms                                                                                                                     10:05:47 AM
^C
```

**Comment by manniL** [comment]
This is most likely an issue with a module you are using, probably `nuxt-scheduler`. Closing as duplicate of https://github.com/nuxt/cli/issues/193#issuecomment-2012957895

**Comment by ppteixeira** [comment]
For those still struggling as I was, there's a workaround.

-  adding a `close` hook on nuxt.config worked for me.

```typescript
// nuxt.config.ts

export default defineNuxtConfig({
   // ...
   
   hooks: {
     close: () => {
       process.exit(0)
     }
   },
})
```

That's because nitro calls the `close` hook after prerendering:

```typescript
// NITRO --> src/runtime/entries/nitro-prerenderer.ts

export const closePrerenderer = () => nitroApp.hooks.callHook("close");
```

Reference: https://github.com/nuxt/cli/issues/169#issuecomment-2119274407

**Comment by MrSunshyne** [comment]
Thanks @ppteixeira I ran into this issue today and your suggestion helped me unblock it. 

I was using latest nuxt content 3.12.0 and nuxt 4.2.1

The CI build hangs and wouldn't proceed.
